common/sstuff.hh

```diff
@@ -345,6 +345,16 @@ public:
     return res;
   }
 
+  //! Reads a block of data from the socket to a block of memory
+  //! Only returns when all the data requested is available
+  int readAll(char *buffer, int bytes)
```

Comment:
any particular reason this is using char and int? it could use std::string and size_t instead. also it should probably return ssize_t instead of int.

Reply:
Well recv needs char* buffer. If I pass a std::string into the function then I'd have to dynamically allocate a char* buffer of the right size in the function to use with recv, and then copy the result into the string. That doesn't seem very efficient.

Reply:
Also the int stuff is simply to match the other functions in this class. I should probably change them all if I'm going to change one